Add celeri

maresb opened the pull request:
**TODO**: adapt to the newer release, and add entry point scripts.

<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

maresb commented:
@aseyboldt and @brendanjmeade, please comment here to attest that you consent to becoming feedstock maintainers once this is merged. (You can opt-out later at any time.)

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/celeri/meta.yaml```) and found it was in an excellent condition.

brendanjmeade commented:
I consent to becoming a feedstock maintainer

aseyboldt commented:
I consent to becoming a feedstock maintainer

brendanjmeade commented:
Well that's no fun.  It looks like it has a recent release:

https://pypi.org/project/gmsh/#history

Is it a pinned version issue?

brendanjmeade commented:
That's a relief!

brendanjmeade commented:
I didn't even know that was possible.  This sounds like a challenge.

maresb commented:
@conda-forge/help-python, ready for review!

This is basically auto-generated by Grayskull and updated for CFEP-25.

Thanks to my co-maintainers @aseyboldt and @brendanjmeade for consenting here.

The astute staged-recipe reviewer will note some non-traditional core PyPI dependencies like `pytest` and `ipykernel`. This is was a deliberate decision.

maresb commented:
The Linux build keeps hanging on the build stage and running out of memory:

```
...
    wheel:                   0.45.1-pyhd8ed1ab_1        conda-forge
    zipp:                    3.23.0-pyhd8ed1ab_0        conda-forge

Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Reloading output folder: ...working... done
##[warning]Free memory is lower than 5%; Currently used: 95.12%
##[warning]Free memory is lower than 5%; Currently used: 95.12%
##[warning]Free memory is lower than 5%; Currently used: 95.12%
##[warning]Free memory is lower than 5%; Currently used: 95.12%
##[warning]Free memory is lower than 5%; Currently used: 95.12%
```

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/celeri/recipe.yaml```) and found some lint.

Here's what I've got...


For **recipes/celeri/recipe.yaml**:

 * ❌ The recipe must have a `build/number` section.

For **recipes/celeri/recipe.yaml**:

 * ℹ️ Recipes should usually depend on `matplotlib-base` as opposed to `matplotlib` so that runtime environments do not require large packages like `qt`.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/17025409396](https://github.com/conda-forge/conda-forge-webservices/actions/runs/17025409396). Examine the logs at this URL for more detail.</sub>

maresb commented:
@conda-forge/help-python, ready for review!

Thanks to my co-maintainers @aseyboldt and @brendanjmeade for consenting in https://github.com/conda-forge/staged-recipes/pull/30780#issuecomment-3188309831 and https://github.com/conda-forge/staged-recipes/pull/30780#issuecomment-3188491574.

The astute staged-recipe reviewer will note some non-traditional core PyPI dependencies like `pytest` and `ipykernel` included upstream. This is was a deliberate decision driven by the current stage of development.

I [couldn't get it to work](https://github.com/conda-forge/staged-recipes/pull/30780#issuecomment-3194567465) with conda-build, so I switched to rattler-build. 

Along the way I fixed some problems in the gmsh feedstock in https://github.com/conda-forge/gmsh-feedstock/pull/103, but that's awaiting review, so I included some workarounds in the current recipe.
